docs/en/restful_api.md

```diff
@@ -7,52 +7,65 @@ python3 -m lmdeploy.serve.openai.api_server ./workspace 0.0.0.0 server_port --in
 ```
 
 Then, the user can open the swagger UI: `http://{server_ip}:{server_port}` for the detailed api usage.
-We provide four restful api in total. Three of them are in OpenAI format. However, we recommend users try
-our own api which provides more arguments for users to modify. The performance is comparatively better.
+We provide five restful api in total. Three of them are in OpenAI format. However, we recommend users try
+our own api `generate` which provides more arguments for users to modify. The performance is comparatively better.
+
+**Note** please, lmdeploy supports maintaining session histories on the server for `generate` api. We disable the
+feature by default.
+
+- On interactive mode, the chat history is kept on the server. In a multiple rounds of conversation, you should set
+  the same `session_id` (can't be -1, it's the default number) to `generate` for requests.
+
+- On normal mode, no chat history is kept on the server.
+
+**Note** please, if you want to launch multiple requests, you'd better set different `session_id` for both
+`/v1/chat/completions` and `/generate` apis. Or, we will set them random values.
 
 ### python
 
-Here is an example for our own api `generate`.
+We have integrated the client-side functionalities of these services into the `APIClient` class. Below are some examples demonstrating how to invoke the `api_server` service on the client side.
+
+If you want to use the `/v1/chat/completions` endpoint, you can try the following code:
+
+```python
+from lmdeploy.serve.openai.api_client import APIClient
+api_client = APIClient('http://{server_ip}:{server_port}')
+model_name = api_client.available_models[0]
+messages = [{"role": "user", "content": "Say this is a test!"}]
+for item in api_client.chat_completions_v1(model=model_name, messages=messages):
+    print(item)
+```
+
+For the `/v1/completions` endpoint, we also offer a special batch inference interface for single clients that can accept a list of `prompt`. If you want to use the `/v1/completions` endpoint, you can try:
```

Comment:
According to [openai's api reference of 'v1/compeletions'](https://platform.openai.com/docs/api-reference/completions/create#completions/create-prompt), `prompt` can be string or an array. 
So, I think we should support batch inference in the same API rather than making another one.